src/policy/rbf.cpp

```diff
@@ -181,3 +181,43 @@ std::optional<std::string> PaysForRBF(CAmount original_fees,
     }
     return std::nullopt;
 }
+
+std::optional<std::string> CheckMinerScores(CAmount replacement_fees,
```

Comment:
some related discussion on estimating mining score of the replacement: https://github.com/bitcoin/bitcoin/pull/26451#issuecomment-1303889425

Reply:
This incentive deficiency linked above should be a non-issue for CheckMinerScores since V3 package RBF is a restricted topology which disallows an multiple ancestors entirely.

The two cases of "adding a new unconfirmed input" during RBF with this PR:

1) Package size of two(no dedup), new CPFP child is RBF'ing another (set of) tx, goes through CheckMinerScores
2) Package size of 1(deduped parent), transaction is rejected via single-tx-acceptance rules currently in master. Perhaps a constrained V3-only version of #26451 could remove the pin?

The asymmetry above is annoying and a mild pinning vector.

As of this PR, wallet authors can avoid this pinning scenario entirely by not moving fee utxos between different unconfirmed parents, but it would be nice if asymmetry was avoided to make wallet reasoning simpler.

Reply:
Do I understand correctly that you are pointing out that package RBF and single tx RBF have different rules? And suggesting we try to make them more symmetrical by applying something similar in the single tx rules?

Reply:
Ideally yes. Doesn't have to be done now, it just reduces the utility of new unconfirmed inputs to randomly working sometimes, basically.